tests/src/test/scala/packages/websocket/WebSocketTests.scala

```diff
@@ -17,81 +17,141 @@
-                activation.response.result shouldBe Some(JsObject(
-                    "error" -> "You must specify a payload parameter.".toJson))
-        }
+  it should "Require a payload parameter" in {
+    val run = wsk.action.invoke(websocketSendAction, Map("uri" -> serverURI.toString.toJson))
+    withActivation(wsk.activation, run) { activation =>
+      activation.response.success shouldBe false
+      activation.response.result shouldBe Some(JsObject("error" -> "You must specify a payload parameter.".toJson))
     }
+  }
 
-    it should "Require a uri parameter" in {
-        val run = wsk.action.invoke(websocketSendAction, Map("payload" -> "This is the message to send".toJson))
-        withActivation(wsk.activation, run) {
-            activation =>
-                activation.response.success shouldBe false
-                activation.response.result shouldBe Some(JsObject(
-                    "error" -> "You must specify a uri parameter.".toJson))
-        }
+  it should "Require a uri parameter" in {
+    val run = wsk.action.invoke(websocketSendAction, Map("payload" -> "This is the message to send".toJson))
+    withActivation(wsk.activation, run) { activation =>
+      activation.response.success shouldBe false
+      activation.response.result shouldBe Some(JsObject("error" -> "You must specify a uri parameter.".toJson))
     }
+  }
+
+  override def beforeAll() {
+
+    val websocketServer =
+      """
+         | const WebSocket = require("ws");
+         | const wss = new WebSocket.Server({ port: 8080 });
+         | wss.on("connection", function connection(ws) {
+         |  ws.on("message", function incoming(message) {
+         |    ws.send(message);
+         |  });
+         | });
+         """.replace("\r", "").stripMargin.lines.mkString
+    println(websocketServer)
+    TestUtils.runCmd(TestUtils.DONTCARE_EXIT, new File("."), "docker", "kill", containerName)
+    TestUtils.runCmd(TestUtils.DONTCARE_EXIT, new File("."), "docker", "rm", containerName)
+    TestUtils.runCmd(
+      0,
+      new File("."),
+      "docker",
+      "run",
+      "-d",
+      "-p",
+      "20002:8080",
+      "--name",
+      containerName,
+      "node",
+      "sh",
+      "-c",
+      s"npm install ws; echo '$websocketServer' | node")
+    sleepUntilContainerRunning()
+
+  }
+
+  def sleepUntilContainerRunning() {
+    var counter = 12
+    var running = false
+    do {
+      counter = counter - 1
+      val isdb2Running = TestUtils
+        .runCmd(
+          TestUtils.DONTCARE_EXIT,
+          new File("."),
+          "docker",
+          "exec",
+          "-t",
+          containerName,
+          "curl",
+          "http://localhost:8080",
+          "--connect-timeout",
+          "5")
+
+      if (isdb2Running.exitCode != 0) {
+        println("sleeping 5 seconds to wait for websocket server")
+        Thread.sleep(5000)
+      } else {
+        running = true
+      }
+    } while (counter > 0 && !running)
+  }
```

Comment:
Can you do an assert here to ensure that running == true and print a message saying that the container didn't start if the assert fails.